lua fails to build when cross-compiling

**Describe the bug**
Due to weird hardcoded CC, AR and RANLIB in Makefiles, lua doesn't like to be cross-compiled. The build crashes, unable to find gcc.

**To Reproduce**
Steps to reproduce the behavior:
1. `nix-build --arg crossSystem '{ system = "armv7l-linux" }' '<nixpkgs>' -A lua`
2. Wait for the build to crash

**Expected behavior**
lua cross-compiles successfuly, like any other package

**Screenshots**
Here's an excerpt from the build log:
```
building '/nix/store/isd9w196grdcwgrs00fi3fv54bx31llp-lua-5.2.4-armv7l-unknown-linux-gnueabihf.drv'...
unpacking sources
unpacking source archive /nix/store/28g8gzsd38c3kli6s0f0p0q9a95s7vgm-lua-5.2.tar.gz
source root is lua-5.2.4
setting SOURCE_DATE_EPOCH to timestamp 1424910717 of file lua-5.2.4/README
patching sources
applying patch /nix/store/jpkm2nm2m9xf9m9v44cr5lw9kx3b2rf7-lua-arch.patch
patching file Makefile
patching file src/luaconf.h
patching file src/Makefile
source root is libzip-1.3.0
updateAutotoolsGnuConfigScriptsPhase
configuring
setting SOURCE_DATE_EPOCH to timestamp 1504368659 of file libzip-1.3.0/man/zip_stat.man
building
patching sources
build flags: SHELL=/nix/store/xfghy8ixrhz3kyy6p724iv3cxji088dx-bash-4.4-p23/bin/bash INSTALL_TOP=/nix/store/j4faq0xc9a875n54w7aczgfhkadkklls-lua-5.2.4-armv7l-unknown-linux-gnueabihf INSTALL_MAN=/nix/store/j4faq0xc9a875n54w7aczgfhkadkklls-lua-5.2.4-armv7l-unknown-linux-gnueabihf/share/man/man1 R=5.2.4 LDFLAGS=-fPIC V=5.2 PLAT=linux CFLAGS=-DLUA_USE_LINUX\ -O2\ -fPIC\ -DLUA_COMPAT_ALL
cd src && make linux V=5.2 R=5.2.4
make[1]: Entering directory '/build/lua-5.2.4/src'
make all SYSCFLAGS="-DLUA_USE_LINUX" SYSLIBS="-Wl,-E -ldl -lreadline"
patching script interpreter paths in test-driver
make[2]: Entering directory '/build/lua-5.2.4/src'
gcc -DLUA_USE_LINUX -O2 -fPIC -DLUA_COMPAT_ALL   -c -o lapi.o lapi.c
/nix/store/xfghy8ixrhz3kyy6p724iv3cxji088dx-bash-4.4-p23/bin/bash: gcc: command not found
make[2]: *** [<builtin>: lapi.o] Error 127
make[2]: Leaving directory '/build/lua-5.2.4/src'
make[1]: *** [Makefile:113: linux] Error 2
make[1]: Leaving directory '/build/lua-5.2.4/src'
make: *** [Makefile:55: linux] Error 2
test-driver: interpreter directive changed from " /bin/sh" to "/nix/store/xfghy8ixrhz3kyy6p724iv3cxji088dx-bash-4.4-p23/bin/sh"
builder for '/nix/store/isd9w196grdcwgrs00fi3fv54bx31llp-lua-5.2.4-armv7l-unknown-linux-gnueabihf.drv' failed with exit code 2
```

**Additional context**
I'm currently trying to build a usable server system for armv7l to put my old Raspberry Pi 2 to good use and reporting bugs with cross-compilation as I go. This is the first one I couldn't fix myself with a simple packageOverride.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 4.19.62, NixOS, 19.09pre186857.239fffc90d7 (Loris)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.2.2`
 - channels(root): `"cleveca22, nixos-19.09pre186857.239fffc90d7"`
 - channels(kisik21): `"home-manager"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
attribute:
   - lua
   - lua5
   - lua51
   - lua52
   - lua53
```

--- a/pkgs/development/interpreters/lua-5/interpreter.nix
+++ b/pkgs/development/interpreters/lua-5/interpreter.nix
@@ -40,14 +40,17 @@ self = stdenv.mkDerivation rec {
     "PLAT=macosx"
   ] else [
     "PLAT=linux"
-  ])
+  ]) ++ (if stdenv.buildPlatform != stdenv.hostPlatform then [
+    "CC=${stdenv.hostPlatform.config}-gcc"
+    "RANLIB=${stdenv.hostPlatform.config}-ranlib"
+  ] else [])
   ;
 
   configurePhase = ''
     runHook preConfigure
 
     makeFlagsArray+=(CFLAGS="-DLUA_USE_LINUX -O2 -fPIC${if compat then " -DLUA_COMPAT_ALL" else ""}" )
-    makeFlagsArray+=(${stdenv.lib.optionalString stdenv.isDarwin "CC=\"$CC\""})
+    makeFlagsArray+=(${stdenv.lib.optionalString stdenv.isDarwin "CC=\"$CC\""}${stdenv.lib.optionalString (stdenv.buildPlatform != stdenv.hostPlatform) " 'AR=${stdenv.hostPlatform.config}-ar rcu'"})
 
     installFlagsArray=( TO_BIN="lua luac" INSTALL_DATA='cp -d' \
       TO_LIB="${if stdenv.isDarwin then "liblua.${version}.dylib" else "liblua.a liblua.so liblua.so.${luaversion} liblua.so.${version}"}" )

